solutions/gold/usaco-696.mdx

```diff
@@ -306,10 +307,122 @@ int main() {
 </CPPSection>
 </LanguageSection>
 
-<IncompleteSection>
+## Solution 3: HLD
 
-Add Centroid Decomp/HLD Solution
+**Time Complexity:** $\mathcal{O}(N\log ^2N)$
+
+Sort the nodes from largest to smallest value, do path updates from each node to the root.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+template <class T> class SumSegmentTree {
```

Comment:
put this in a [codesnip](https://usaco.guide/general/working-mdx#code-blocks-and-code-snippets)